lib/library.rb

```diff
@@ -0,0 +1,48 @@
+require 'yaml'
+require 'date'
+
+class Library
+  attr_accessor :bookrack
+  ALLOWED_LOAN_DAYS = 30
+
+
+  def initialize
+    @bookrack = YAML.load_file('./lib/booksdb.yml')
+  end
+
+  def available_books
+    @bookrack.select{|obj| obj[:available] == true}
+  end
+
+  def view_all_books
+    @bookrack
+  end
+
+  def check_book_availability(book_index)
+    bookrack[book_index].nil? ? book_not_found : checkout_book_from_bookrack(book_index)
+  end
```

Comment:
This `bookrack` is not the same as `@bookrack`... Is it performing the way you expect?